crates/compiler/tests/compilation_basic_add.rs

```diff
@@ -7,8 +7,9 @@ use itertools::Itertools;
 mod common;
 
 #[test]
-fn compiles_add() -> anyhow::Result<()> {
+fn compiles_add() -> miette::Result<()> {
     // We start by constructing and running the compiler
+    common::set_miette_reporting()?;
```

Comment:
(Do we want this on for the test?)

Reply:
Yes! In the case that something errors we will get decent diagnostics.